tock/tock/management/commands/deactivate_stale_users.py

```diff
@@ -0,0 +1,35 @@
+from datetime import datetime, timedelta, timezone
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from rest_framework.authtoken.models import Token
+
+User = get_user_model()
+
+@transaction.atomic
+class Command(BaseCommand):
+    help = 'Deactivate user when they have not logged in for a set amount of time'
+
+    def add_arguments(self, parser):
+        parser.add_argument('days_not_logged_in', nargs='?', type=int, default=90)
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        now = datetime.now(timezone.utc)
+
+        past_time = now - timedelta(days=options['days_not_logged_in'])
+        user_list = User.objects.filter(last_login__lte=past_time, is_active=True)
+        user_count = 0
+        token_count = 0
+        for user in user_list:
+          user.is_active = False
+          user.save()
+          user_count += 1
+
+          token = Token.objects.filter(user=user)
```

Comment:
Each user can only have on token.  I was debating to use `get()` or `filter()`.  I settled with `filter()` because it looks a bit cleaner.  If I use `get()`, I will have to do a `try`/`except` block, where the `except Token.DoesNotExist` will do nothing.  Also performance-wise, it is probably similar because I am getting a token based on user (foreign key involved) here, and underneath, `get()` uses `filter()` here.

Let me know if I am missing something here though.

Reply:
I think the only thing is that when I see filter() I inherently expect to see some iteration through the results. Your short-circuit logic here should work fine given the constraints. It looked a little odd to me, but as you describe it, I think it's fine.

Reply:
I see your point, I will put in a comment to clarify.